page-objects/pages/TASKListPage.js

```diff
@@ -136,6 +141,15 @@ async function clickDeleteTaskButtonInPopup() {
 	await waitHelpers.waitUntilAjaxIsFinished();
 }
 
+async function clickTaskWithoutDuedate(){
+	await elementHelpers.clickAndWait(clickWithoutDueDate);
+}
+
+async function clickUngradedTask(){
+	await elementHelpers.clickAndWait(ungradedTaskSection);
+	await elementHelpers.clickAndWait(ungradedTask);
```

Comment:
I'm thinking of dividing this into two seperate steps. One for going to the ungraded tasks section and one for actually opening the ungraded task.

Reply:
Agree. Please split it up to functions like clickUngradedTaskSection and clickUngradedTask. You will need two function calls, one after another, in ..-steps.js file